src/server/handlers/universal.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { match, RouterContext } from 'react-router';
+import { renderToString } from 'react-dom/server';
+
+import Html from '../helpers/html';
+import conf from '../configure.js';
+import configureStore from '../../common/store/configureStore';
+
+let routes = require('../../common/routes').default;
+
+import assets from '../../../webpack-assets.json';
+
+export const universal = (req, res) => {
+  if (process.env.NODE_ENV === 'development') {
+    // Hot-reload application files when changes
+    // made when running as a development site
+    routes = require('../../common/routes').default;
+  }
+
+  match({ routes, location: req.url }, (error, redirectLocation, renderProps) => {
+    handleMatch(req, res, error, redirectLocation, renderProps);
+  });
+};
+
+export const handleMatch = (req, res, error, redirectLocation, renderProps) => {
+  if (error) {
+    res.status(500).send(error.message);
+  } else if (redirectLocation) {
+    res.redirect(302, redirectLocation.pathname + redirectLocation.search);
+  } else if (renderProps) {
+
+    const initialState = {};
+
+    if (req.session) {
+
+      if (req.session.authenticated) {
+        initialState['identity'] = {
+          isAuthenticated: req.session.authenticated,
+          name: req.session.name,
+          email: req.session.email
+        };
+      }
+
+      if (req.session.error) {
+        initialState['oyez'] = [{
+          'message': req.session.error,
+          'status': 'error'
+        }];
+
+        delete req.session.error;
+      }
+
+    }
+
+    const store = configureStore(initialState);
+
+    // config we share from server side to client side
+    const config = {
+      UNIVERSAL: conf.get('UNIVERSAL')
```

Comment:
I think we wanted to get rid of UNIVERSAL/SERVER prefixes in config.
But maybe it's something for another branch.